concept/thing/thing_manager.rs

```diff
@@ -2931,35 +2920,57 @@ impl ThingManager {
     //   (create role type, set cardinality annotation, unset cardinality annotation, ...)
     // * Clean up all parts of a relation index to do with a specific role player
     //   after the player has been deleted.
-    pub(crate) fn relation_index_player_deleted(
+    pub(crate) fn relation_index_player_links_unset(
         &self,
         snapshot: &mut impl WritableSnapshot,
         relation: Relation,
         player: impl ObjectAPI,
         role_type: RoleType,
+        is_index_persisted: bool, // An optimization to avoid another storage lookup
```

Comment:
This only signifies that `links` was persisted. `index` may or may not have been.